modular_citadel/code/modules/nifsoft/nif.dm

```diff
@@ -1,8 +1,857 @@
+	repair_multiplier = 0.05 //Some people be desperate.
+
+/obj/item/device/nif/authentic //KHI removed, still the "best" you can get, minus special features
 	name = "\improper Vey-Med NIF"
-	desc = "A genuine Vey-Med nanotechnology fabricator, now the size of a shiny antique stamp. Far more durable than any knockoffs on the market."
+	desc = "Vey-Med's entry into the NIF market. Usually reserved for only the rich."
+	quality = 4
+	durability = 1000
+	repair_multiplier = 5
+
+/obj/item/device/nif/specialty
+	name = "\improper specialty NIF"
+	desc = "A high-endurance NIF with better nanites and programming. Produced by specialty manufacturers."
+	quality = 3
+	durability = 500
+	charge_use_multiplier = 1.25
+
+/obj/item/device/nif/bioadap
+	name = "adaptive NIF"
+	desc = "An intelligent NIF that can adapt certain NIFSofts for use outside of their intended audience. It's not perfect in this regard, however."
+	quality = 3
+	durability = 25
+	bioadap = TRUE
+	nif_flags = NIF_FORCE_INSTALL
+	charge_use_multiplier = 1.1
+	repair_multiplier = 0.5
+
+/obj/item/device/nif/quality
+	name = "quality NIF"
+	desc = "A well-programmed NIF that can support higher-quality software than usual."
+	quality = 4
+	durability = 100
+
+/obj/item/device/nif/durable
+	name = "durable NIF"
+	desc = "A durably-built NIF."
+	quality = 2
+	durability = 1000
+
+/obj/item/device/nif/sandbox
+	name = "sandboxed NIF"
+	desc = "This NIF sandboxes all NIFSoft operation, and temporarily disables them for an instant every few seconds. \
+	Allows removing Compliance Disks."
+	quality = 2 //Gives you compliance-level programs without the risks of compliance disks.
+	durability = 50 //But is less durable because of it.
+	nif_flags = NIF_FORCE_DELETE
+
+/obj/item/device/nif/backup
+	name = "survivalist NIF"
+	desc = "This NIF is preloaded with robust mind backup and soulcatcher utilities, and runs on a minimum of power."
+	quality = 2
+	durability = 40 //Not durable, but it does its best to keep you alive.
+	install_blind = 1 MINUTES //Really fast because this is for paranoid people.
+	install_synchronize = 20 MINUTES //Not so fast but no side-effects.
+	install_side_effect_chance = 100
+	charge_use_multiplier = 0.5 //Hunger Begone
+	repair_multiplier = 8 //Only costs 5 units of nif repair stuff to fully repair. Mainly for speed of repair.
+
+	load_starting_software()
+		new /datum/nifsoft/mindbackup(src)
+		new /datum/nifsoft/soulcatcher(src)
+		. = ..()
+
+// Cheap NIFs
+/obj/item/device/nif/consumer //Buildable in Autolathe
+	name = "consumer NIF"
+	desc = "A mass-production NIF that can only run simple programs. Lasts practically forever. Installs almost instantly."
+	quality = 0
+	durability = 500
+	install_blind = 1 MINUTES
+	install_synchronize = 3 MINUTES
+	charge_use_multiplier = 0.1
+	matter = list(DEFAULT_WALL_MATERIAL = 2000)
+	repair_multiplier = 5
+
+/obj/item/device/nif/cheap //Needs only metal and glass to be built once unlocked in Research
```

Comment:
This should just be the consumer NIF from cargo. Much less bloaty.

Reply:
Consumer NIF can only install huds. Cheap can install both huds and vision augments like mesons.